packages/playwright/src/reporters/list.ts

```diff
@@ -109,11 +117,10 @@ class ListReporter extends BaseReporter {
     if (!this._printSteps) {
       if (isTTY)
         this._updateLine(this._testRows.get(test)!, colors.dim(formatTestTitle(this.config, test, step.parent)) + this._retrySuffix(result), this._testPrefix(testIndex, ''));
-      return;
```

Comment:
Shouldn't we leave this line now? It seems like when `!this._printSteps`, it will update the current line, and then *also* print the step information? I would think it should not print steps in this case.

I also wonder why all the tests pass. Do we miss coverage of non-printSteps mode?

Reply:
Alright, so I think that https://github.com/Skn0tt/playwright/blob/dc152ad614cc7f83bc92ec5bc8d8b97238442d5a/tests/playwright-test/reporter-list.spec.ts#L94-L127 perfectly covers non-printSteps. The reason we didn't see this regression in tests is that in https://github.com/Skn0tt/playwright/blob/dc152ad614cc7f83bc92ec5bc8d8b97238442d5a/packages/playwright/src/reporters/list.ts#L134, `this._stepRows.get(step)!` is undefined. That leads to the lines being prefixed with `undefined :`, and then https://github.com/Skn0tt/playwright/blob/dc152ad614cc7f83bc92ec5bc8d8b97238442d5a/tests/playwright-test/reporter-list.spec.ts#L110 filters that out. I'll see if we can work around that somehow.

Reply:
In https://github.com/microsoft/playwright/pull/31703/commits/91f8c7d5ec755ec176ae6750805f057938e7cdc5, i've prefixed the line numbers with `#`, and updated the regex us to use that for matching. This ensures that lines with an `undefined` row number aren't swallowed, see here:
<img width="1728" alt="Screenshot 2024-07-16 at 19 58 54" src="https://github.com/user-attachments/assets/17521c61-9a0e-49cd-a7a8-29f3206aed2a">